status/health/window/key_error_source.go

```diff
@@ -149,27 +160,31 @@ func (m *multiKeyHealthyIfNotAllErrorsSource) HealthStatus(ctx context.Context)
 	m.sourceMutex.Lock()
 	defer m.sourceMutex.Unlock()
 
-	m.pruneOldKeys(m.errorStore, m.lastError)
-	m.pruneOldKeys(m.successStore, nil)
-
-	params := make(map[string]interface{})
-	for _, key := range m.errorStore.List().Keys() {
-		if _, hasSuccess := m.successStore.Get(key); hasSuccess {
-			continue
+	var healthCheckResult health.HealthCheckResult
+	if m.requireFirstFullWindow && m.timeProvider.Now().Sub(m.startTime) < m.windowSize {
+		healthCheckResult = whealth.HealthyHealthCheckResult(m.checkType)
```

Comment:
Yes that's more intuitive this was a holdover from the anchored windows used in the single source version of this. I'll update both so that non-anchored versions of both that require the first full windows will report REPAIRING for only errors in their initial windows.